ondemand/install.sh

```diff
@@ -7,6 +7,185 @@ log_info() {
+    --with-http_flv_module \
+    --with-http_mp4_module \
+    --with-http_gunzip_module \
+    --with-http_gzip_static_module \
+    --with-http_random_index_module \
+    --with-http_secure_link_module \
+    --with-http_degradation_module \
+    --with-http_stub_status_module \
+    --with-mail \
+    --with-mail_ssl_module \
+    --with-pcre \
+    --with-pcre-jit \
+    --add-module=../src/nginx_module \
+    --with-cc-opt="$NGINX_CCOPTS" \
+    --with-ld-opt="$LDOPTS" \
+    --with-debug
+
+  make -j$(nproc) && make install INSTALLDIRS=vendor
+  mkdir -p  $NGINX_DATADIR/html
+  mkdir -p  $NGINX_CONFDIR
+  mkdir -p  $NGINX_HOME
+  mkdir -p  $NGINX_HOME_TMP
+  mkdir -p  $NGINX_LOGDIR
+
+  set -x
+  cd "$BUILD_DIR/passenger-$PASS_VERSION"
+  which ruby
+	rake fakeroot \
+	    NATIVE_PACKAGING_METHOD=rpm \
+	    FS_PREFIX=$PREFIX \
+	    FS_BINDIR=$PREFIX/bin \
+	    FS_SBINDIR=$PREFIX/sbin \
+	    FS_DATADIR=$PREFIX/usr/share \
+	    FS_LIBDIR=$PREFIX/lib64 \
+	    FS_DOCDIR=$PREFIX/usr/share/doc \
+	    RUBY=$(which ruby) \
+	    RUBYLIBDIR=$RUBY_LIBDIR \
+	    RUBYARCHDIR=$RUBY_LIBDIR \
+	    APACHE2_MODULE_PATH=$PREFIX/usr/lib/apache2/modules/mod_passenger.so \
+			OPTIMIZE=yes \
+			CACHING=false \
+			EXTRA_CFLAGS="$PASSENGER_CCOPTS" \
+			EXTRA_CXXFLAGS="$PASSENGER_CCOPTS"
+
+  cp -a $BUILD_DIR/passenger-$PASS_VERSION/pkg/fakeroot/* /
+  cd $BUILD_DIR/passenger-$PASS_VERSION
+  ./dev/install_scripts_bootstrap_code.rb --ruby $RUBY_LIBDIR \
+	  $PREFIX/bin/passenger* \
+	  $PREFIX/sbin/passenger* \
+	  `find $PREFIX -name rack_handler.rb`
+
+  ./dev/install_scripts_bootstrap_code.rb --nginx-module-config $PREFIX/bin $PREFIX/usr/share/passenger/ngx_http_passenger_module/config
+  chmod +x $PREFIX/usr/share/passenger/helper-scripts/wsgi-loader.py
+}
+
+install_os_deps() {
+  dnf -y update && \
+    dnf install -y dnf-utils epel-release && \
+    dnf config-manager --set-enabled powertools && \
+    dnf -y module enable nodejs:12 ruby:2.7 && \
+    dnf install -y \
+        file lsof sudo gcc gcc-c++ git \
+        patch lua-posix rsync ruby ruby-devel python2 python3 \
+        nodejs sqlite sqlite-devel nmap-ncat httpd httpd-devel mod_ssl \
+        libcurl-devel autoconf openssl-devel jansson-devel libxml2-devel \
+        libxslt-devel gd-devel
+  gem install rake dotenv bcrypt
+}
+
+build_ood_src() {
+  cd $BUILD_DIR
+
+  git clone https://github.com/cisco/cjose
+  cd cjose
+  git checkout 0.6.1
+  ./configure
+  make && make install
+
+  # so mod_auth_openidc can find cjose
+  export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig
+
+  cd $BUILD_DIR
+  git clone https://github.com/zmartzone/mod_auth_openidc.git
+  cd mod_auth_openidc
+  git checkout v2.4.9.4
+  sh autogen.sh
+  ./configure
+  make && make install
+
+  cd $BUILD_DIR
+  OOD_VERSION='2.0.27'
+  wget "https://github.com/OSC/ondemand/archive/refs/tags/v$OOD_VERSION.tar.gz"
+  tar -xf "v$OOD_VERSION.tar.gz"
+  cd ondemand-$OOD_VERSION
+  bundle config --local path ~/vendor/bundle
```

Comment:
Looks like Nokogiri gem includes pre-compiled binaries by default. The binaries built for arm64 require `GLIBC_2.29` which is not in rockylinux8. So we need to add these 2 lines here to force Nokogiri to build from source:

```diff
+  bundle config build.nokogiri --use-system-libraries
+  bundle config set force_ruby_platform true
```

For reference, here's the error I was getting: 

```
#0 7833.1 ERROR: It looks like you're trying to use Nokogiri as a precompiled native gem on a system with glibc < 2.17:
#0 7833.1 
#0 7833.1   /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/gems/gems/nokogiri-1.13.6-aarch64-linux/lib/nokogiri/2.7/nokogiri.so) - /usr/share/gems/gems/nokogiri-1.13.6-aarch64-linux/lib/nokogiri/2.7/nokogiri.so
#0 7833.1 
#0 7833.1   If that's the case, then please install Nokogiri via the `ruby` platform gem:
#0 7833.1       gem install nokogiri --platform=ruby
#0 7833.1   or:
#0 7833.1       bundle config set force_ruby_platform true
#0 7833.1 
#0 7833.1   Please visit https://nokogiri.org/tutorials/installing_nokogiri.html for more help.
#0 7833.1 
#0 7833.1 rake aborted!
#0 7833.1 LoadError: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/gems/gems/nokogiri-1.13.6-aarch64-linux/lib/nokogiri/2.7/nokogiri.so) - /usr/share/gems/gems/nokogiri-1.13.6-aarch64-linux/lib/nokogiri/2.7/nokogiri.so
```